tests/unit/projection-test.js

```diff
@@ -2224,6 +2235,82 @@ module('unit/projection', function(hooks) {
       assert.equal(get(projectedPreview, 'title'), BOOK_TITLE);
     });
 
+    skip(
+      'Unloading a record removes it from record arrays, which have reference to it',
+      function(assert) {
+        // we need additional records to be able to resolve the references
+        run(() => {
+          this.store.push({
+            data: {
+              id: OTHER_BOOK_ID,
+              type: BOOK_CLASS_PATH,
+              attributes: {},
+            },
+          });
+        });
+
+        run(() => {
+          this.store.push({
+            data: {
+              id: OTHER_BOOK_ID,
+              type: BOOK_PREVIEW_PROJECTION_CLASS_PATH,
+            },
+          });
+        });
+
+        let { baseModel, projectedPreview } = this.records;
+
+        // load the record arrays
+        let booksInSeriesBase = get(baseModel, 'otherBooksInSeries');
+        let booksInSeriesProjectedPreview = get(
+          projectedPreview,
+          'otherBooksInSeries'
+        );
+        let otherProjectedPreview = get(
+          booksInSeriesProjectedPreview,
+          'firstObject'
+        );
+
+        // precondition
+        assert.equal(
+          get(booksInSeriesBase, 'length'),
+          1,
+          'Expected otherBooksInSeries length to be one for base'
+        );
+        assert.equal(
+          get(booksInSeriesProjectedPreview, 'length'),
+          1,
+          'Expected otherBooksInSeries length to be one for projected preview'
+        );
+
+        // unload a projection referenced in a record array
+        run(() => {
+          otherProjectedPreview.unloadRecord();
+        });
+
+        assert.equal(
+          get(booksInSeriesBase, 'length'),
+          1,
+          'Expected otherBooksInSeries length to be unchanged for base'
+        );
+        assert.equal(
+          get(booksInSeriesProjectedPreview, 'length'),
+          1,
```

Comment:
why don't we expect it to be removed?

Reply:
Because if the record is not available (`peekRecord` returns `undefined`/`null`) in the first place, when the record array is created, it will leave a whole in the array instead of shortening the record array. This behavior is defined in the `resolveRecordArrayInternalModels` function.
We should expect the same behavior if the code forcibly unload a record and cause `peekRecord` to return `undefined`/`null`.

Reply:
so is this the case where we know a record should exist but it simply has not been loaded?

Reply:
Well, we assume it should exist, but it does not. There are several cases, which may lead to this and the application will have to expect these case and handle them accordingly.

Reply:
@runspired m3 doesn't support async relationships (which is more or less what this is).

The more consistent thing to do here is to assert.